enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,258 @@
+---
+title: windows-node-health-management
+authors:
+  - "@sebsoto"
+  - "@saifshaikh48"
+reviewers:
+  - "@openshift/openshift-team-windows-containers"
+approvers:
+  - "@aravindhp"
+  - "@sdodson"
+creation-date: 2021-12-16
+last-updated: 2021-12-16
+status: implementable
+---
+
+# Windows Node Health Management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of this enhancement is to ensure that all Windows nodes managed by the Windows Machine Config Operator (WMCO)
+maintain a usable state. With health management implemented, WMCO will automatically ensure that OpenShift related services 
+running on the underlying Windows VMs have a status and runtime configuration inline with the expectations of the installed operator version.
+
+## Motivation
+
+Currently [WMCO](https://github.com/openshift/windows-machine-config-operator/) properly fulfills its goal of
+configuring Windows VMs into Nodes. Once the Node objects exists, barring specific conditions, WMCO does not have the
+functionality to detect and correct changes in Node state. Windows nodes should be resilient enough to not require
+manual intervention when they encounter small issues that may prevent expected functionality. This work serves as
+part of a larger initiative to minimize the downtime of customer workloads.
+
+### Goals
+
+* Ensuring WMCO-managed Windows services on the VMs associated with Windows Node objects are in the expected state and configured with the expected values:
+  + Kubernetes components
+  + node-level networking components
+  + container runtime
+  + metrics collector/exporter
+* Consolidating the Windows Machine Config Bootstrapper ([WMCB](https://github.com/openshift/windows-machine-config-bootstrapper))
+  module into the [WMCO](https://github.com/openshift/windows-machine-config-operator/) project, maintaining the
+  existing ability to properly configure Windows VMs to join the cluster as Windows Nodes.
+
+
+### Non-Goals
+
+* Diagnosing or remediating generic Windows issues, unrelated to changes made by WMCO.
+* Cluster-level resource management, such as deleting or re-creating WMCO-configured Machines that enter an unrecoverable state (this is a responsibility of [MachineHealthChecks](https://docs.openshift.com/container-platform/latest/machine_management/deploying-machine-health-checks.html)).
+
+
+## Proposal
```

Comment:
The proposal should state the overall workflow and not go into design details which is why the `Design` section is empty.

Reply:
Is this implying moving the discussion around combining WMCO/WMCB into the Design Details section?